### PR TITLE
Refactor FormSubmitError for easier testing, better error display

### PR DESCRIPTION
**Pull request by keithjgrant**
##### SUMMARY
Reworks how FormSubmitError parses error messages from an API response:

* parsing is now done entirely in a util function, which can be test-driven for any future changes
* No longer display field error messages both below the field and beneath the whole form
* If the API returns a field error for a field that Formik doesn't have a value for (e.g. `rrule` in the Schedules form), the error is not set on Formik and instead displays at the bottom of the form

addresses #7515 

This also addresses #7822, which was caused by sending errors to PatternFly form components as arrays rather than strings

##### ISSUE TYPE
 - Improvement

##### COMPONENT NAME
 - UI

### CODE REVIEW

File: awx/ui_next/src/components/FormField/sortErrorMessages.test.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import sortErrorMessages from './sortErrorMessages';
+
+describe('sortErrorMessages', () => {
+  let consoleError;
+  beforeEach(() => {
+    consoleError = global.console.error;
```
Comment (jakemcdermott) [inline awx/ui_next/src/components/FormField/sortErrorMessages.test.js:7]: I recommend a comment here because it's not entirely apparent why this is needed (until you go looking in the utility function)

File: awx/ui_next/src/components/FormField/FormSubmitError.jsx [hunk 2]
```diff
@@ -2,62 +2,21 @@ import React, { useState, useEffect } from 'react';
...
 
 function FormSubmitError({ error }) {
   const [errorMessage, setErrorMessage] = useState(null);
-  const { setErrors } = useFormikContext();
+  const { values, setErrors } = useFormikContext();
 
   useEffect(() => {
-    if (!error) {
-      return;
+    const { formError, fieldErrors } = sortErrorMessages(error, values);
```
Comment (jakemcdermott) [inline awx/ui_next/src/components/FormField/FormSubmitError.jsx:12]: :+1: nice. I like that we're pulling this into a separate module/function for testing.

**Review by jakemcdermott — APPROVED** [review]
lgtm

**Comment by nixocio** [comment]
Just to link issues related to the same part of code: https://github.com/ansible/awx/issues/7822

**Comment by keithjgrant** [comment]
QE: this PR modifies how API errors display on nearly all forms in UI-next. It's worth testing multiple different forms to be sure errors display correctly (this does not change client-side validation errors, however, only errors returned by the API)

**Comment by keithjgrant** [comment]
Bug: Nested field values (like the one shown in 7822) don't display error message properly. Working on getting that fixed now

_edit_: this is now fixed

File: awx/ui_next/src/components/FormField/sortErrorMessages.test.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import sortErrorMessages from './sortErrorMessages';
+
+describe('sortErrorMessages', () => {
+  let consoleError;
+  beforeEach(() => {
+    // Component logs errors to console. Hide those during testing.
+    consoleError = global.console.error;
+    global.console.error = () => {};
```
Comment (AlexSCorey) [inline awx/ui_next/src/components/FormField/sortErrorMessages.test.js:8]: I'm so glad you found a way to do this. Huge fan!

File: awx/ui_next/src/components/FormField/sortErrorMessages.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    Object.keys(error.response.data).forEach(fieldName => {
+      const errors = error.response.data[fieldName];
+      if (!errors) {
+        return;
+      }
+      const errorsArray = Array.isArray(errors) ? errors : [errors];
+      if (typeof formValues[fieldName] === 'undefined') {
+        formErrors = [...formErrors, ...errorsArray];
+      } else {
+        fieldErrors[fieldName] = errorsArray.join('; ');
```
Comment (AlexSCorey) [inline awx/ui_next/src/components/FormField/sortErrorMessages.js]: Will this display the error message with a `;` at the end of it under the form field?

**Review by AlexSCorey — COMMENTED** [review]
Just seeking some clarity for 1 line of the code.

File: awx/ui_next/src/components/FormField/sortErrorMessages.js (hunk 4 above)
Comment (keithjgrant) [reply awx/ui_next/src/components/FormField/sortErrorMessages.js]: If there are multiple errors, it will put `; ` between each of them

**Review by jakemcdermott — APPROVED** [review]
:ship:

File: awx/ui_next/src/components/FormField/sortErrorMessages.test.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    expect(parsed).toEqual({
+      formError: 'Other stuff error',
+      fieldErrors: {
+        inputs: {
+          url: 'URL Error',
+        },
+      },
+    });
+  });
+});
```
Comment (jakemcdermott) [inline awx/ui_next/src/components/FormField/sortErrorMessages.test.js:146]:
```suggestion
  test('should return null fieldErrors and blank form error when error contains unhandled error type', () => {
    const error = { response: { data: { unhandled: 9000 } } };
    const formValues = { unhandled: '' };
    const parsed = sortErrorMessages(error, formValues);
    expect(parsed).toEqual({ fieldErrors: null, formError: '' });
  });
});
```

Comment (jakemcdermott) [reply awx/ui_next/src/components/FormField/sortErrorMessages.test.js:146]: @keithjgrant Definitely an edge case (and probably unlikely to happen) but it seemed like the type of nuanced behavior that we'd end up implicitly depending on in a form somewhere, someway, somehow, so I propose we add this test case to get the branch coverage of the `sortErrorMessages` module to :100:

**Review by jakemcdermott — APPROVED** [review]
@keithjgrant Looks good to me! Given all the different data structures we may need to process when handling errors,  I _really_ like all the additional test coverage you're adding :+1: :+1:  This will be tremendously helpful.

I have a suggested addition to the tests:
https://github.com/ansible/awx/pull/8339/files#r503979636

**Comment by unlikelyzero** [comment]
@keithjgrant do you mind just running yolok? If that passes, this can merge, but please leave this https://github.com/ansible/awx/issues/7822 in needs test

**Comment by jakemcdermott** [comment]
regate

**Comment by unlikelyzero** [comment]
regate

**Comment by unlikelyzero** [comment]
Note: this has broken the https://dashboard.cypress.io/projects/ui2mq7/runs/3046/failures/ebc9ffef-fcd3-45e9-99df-e1cb5eb531d0/screenshots Error handling locator on this test.

Regating now to get it in before the nightly visual regression run
